packages/backend/src/modules/tvl2/indexers/ChainAmountIndexer.ts

```diff
@@ -35,41 +35,38 @@ export class ChainAmountIndexer extends ManagedMultiIndexer<ChainAmountConfig> {
     to: number,
     configurations: UpdateConfiguration<ChainAmountConfig>[],
   ): Promise<number> {
-    const timestamp = this.$.syncOptimizer.getTimestampToSync(
-      new UnixTime(from),
-    )
+    const configurationsToSync = configurations.filter((c) => !c.hasData)
 
-    if (timestamp.gt(new UnixTime(to))) {
-      return Promise.resolve(to)
+    if (configurationsToSync.length !== configurations.length) {
+      this.logger.info('Filtered out configurations with data', {
+        from,
+        to,
+        skippedConfigurations:
+          configurations.length - configurationsToSync.length,
+        configurationsToSync: configurationsToSync.length,
+      })
     }
 
-    const configurationsWithMissingData = configurations.filter(
-      (c) => !c.hasData,
-    )
-
-    if (configurationsWithMissingData.length !== configurations.length) {
-      this.logger.info('Skipping update for configurations with data', {
-        configurations: configurations.length,
-        configurationsWithMissingData: configurationsWithMissingData.length,
+    if (configurationsToSync.length === 0) {
+      this.logger.info('No configurations to sync', {
+        from,
+        to,
       })
+      return to
     }
 
-    const blockNumber =
-      await this.$.blockTimestampsRepository.findByChainAndTimestamp(
-        this.$.chain,
-        timestamp,
-      )
-    assert(blockNumber, 'Block number not found')
+    const timestamp = this.$.syncOptimizer.getTimestampToSync(from, to)
+    const blockNumber = await this.getBlockNumber(timestamp)
 
     const amounts = await this.$.amountService.fetchAmounts(
-      configurationsWithMissingData,
-      blockNumber.blockNumber,
       timestamp,
+      blockNumber,
+      configurationsToSync,
     )
 
     this.logger.info('Fetched amounts for timestamp', {
       timestamp: timestamp.toNumber(),
-      blockNumber: blockNumber.blockNumber,
+      blockNumber: blockNumber,
```

Comment:
O lines of code saved
```suggestion
      blockNumber,
```